update LMS docs (#18323)
cockroachdb/docs: src/current/v23.2/live-migration-service.md

This section describes the most important and commonly used values. For details on all configurable values, refer to the [`values.yaml`](https://github.com/cockroachdb/molt-helm-charts/blob/main/lms/values.yaml) file.

#### LMS version

~~~ yaml
image:
  tag: 0.2.4  
~~~

> `image.tag` specifies the LMS version. This **must** match the installed [`molt-lms-cli`](#molt-lms-cli) version.

Reviewer: Might worth noting the `molt-lms-cli` version can be queried with `molt-lms-cli version`

In the [Helm configuration](#configuration), `orchestrator.sslVolumes` and `orchestrator.sslVolumeMounts` must specify [volumes](https://kubernetes.io/docs/concepts/storage/volumes/#secret) and mount paths that contain the server-side certificates. The path to each file is specified as an environment variable in `orchestrator.env`. Cockroach Labs recommends mounting certificates to `/app/certs`.

{{site.data.alerts.callout_info}}

> Certificates **must** be mounted in a readable format, or the LMS will error. The format should match the output of `cat {certificate}`.

Reviewer 2: ```suggestion
Certificates **must** be mounted in a readable format, or the LMS will error. The format should match the output of `cat {certificate}` on your host machine.
```
Maybe just add this for specificity.

{% comment %}
| `cutover immediate`  | Initiate an [immediate cutover](#immediate-cutover). This switches the source of truth to the target database. For usage details, see [Immediate cutover](#immediate-cutover). |
{% endcomment %}

#### Subcommands

The following subcommands are run after the `cutover consistent` command.

|   Subcommand   |                                                                                                                               Usage                                                                                                                               |
|----------------|-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
| `begin`        | Begin a consistent cutover. This pauses traffic to the source database. See additional [flags](#cutover-consistent-begin-flags) and [example](#consistent-cutover).                                                                                                                  |

> | `commit`       | Commit a consistent cutover. This resumes traffic on the target database. This is only effective after running `cutover consistent begin`. See [example](#consistent-cutover).                                                                                    |

Reviewer: ```suggestion
| `commit`       | Commit a consistent cutover. This resumes traffic on the target database. This is only effective after running `cutover consistent begin`. See [example](#consistent-cutover).                                                                                    | `commit`       | Commit a consistent cutover. This resumes traffic but send it to the target database, meaning the source of truth is now flipped. This is only effective after running `cutover consistent begin`. See [example](#consistent-cutover).
```

> | `abort`        | Abort a consistent cutover after running `cutover consistent begin`, unless you have also run `cutover consistent commit`. This resumes traffic to the source database.                                                                                           |

Reviewer: I wonder if it worth adding `unless you have also run commit`.

Author: This is here already!

#### `cutover consistent begin` flags

|           Flag          |                                                                                                                                                                                                                      Description                                                                                                                                                                                                                      |
|-------------------------|-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|

> | `--abort-begin-timeout` | Maximum duration to wait before the LMS aborts cutover following a `ctrl-c` command.<br><br>**Default:** `"2s"`                                                                                                                                                                                                                                                                                                                                       |

Reviewer: ```suggestion
| `--abort-begin-timeout` | Maximum duration of the orchestrator to wait for confirmation from all LMS proxies that the cutover is aborted, upon the orchestrator receives an abort command from a `ctrl-c` command via the `molt-lms-cli`. This affects the performance of the `ctrl-c` command only, and user is not expected to configure this setting frequently. <br><br>**Default:** `"2s"`  
```

> | `--begin-timeout`       | Maximum duration to wait before traffic to the LMS is paused for consistent cutover. This should be the approximate length of the longest-running transaction, e.g., `30s`. If no `--begin-timeout` value is specified, the LMS waits for all transactions and queries to finish before pausing traffic.                                                                                                                                              |

Reviewer: ```suggestion
| `--begin-timeout`       | Maximum duration to wait before traffic to the LMS is paused for consistent cutover, i.e. the limit of wait for all connections to become transaction-and-query-free. This should be the approximate length of the longest-running transaction, e.g., `30s`. If no `--begin-timeout` value is specified, the LMS waits for all transactions and queries to finish before pausing traffic.          
```

> | `-l`, `--lms-addresses` | LMS instances to run the command against. This can be a comma-separated list of IP addresses (e.g., `127.0.0.1`), IP addresses and ports (e.g., `127.0.0.1:1024`), or hostnames (e.g., `https://lms.net`). By default, cutover is performed on all LMS instances deployed in the same environment as the orchestrator. LMS addresses should only be manually specified if the orchestrator is deployed in a different environment than the instances. |

Reviewer: ```suggestion
| `-l`, `--lms-addresses` | LMS instances to run the command against. This can be a comma-separated list of IP addresses (e.g., `127.0.0.1`), IP addresses and ports (e.g., `127.0.0.1:1024`), or hostnames (e.g., `https://lms.net`). By default, cutover is performed on all LMS instances deployed in the same environment as the orchestrator. For example, if all the orchestrator and LMS instances are deployed in the same k8s cluster, the addresses of all LMS instances are automatically detected by the orchestrator and no user configuration is needed. LMS addresses should only be manually specified if the orchestrator is deployed in a different environment than the instances. |
```
@ryanluu12345 might want to confirm if this is accurate too.

Reviewer 2: I think we would want to clarify that its in the same Namespace. https://github.com/cockroachlabs/crdb-proxy/blob/main/orchestrator/api/lms_connections_retriever.go#L45 since that is how we run the query.

- Query results from the source of truth are returned to the application.
- Writes must be manually replicated from the source database to the target database.

> You **must** use this mode to perform a [consistent cutover](#consistent-cutover), along with a database replication technology that replicates writes to the target database.

Reviewer: ```suggestion
You **must** use the `none` mode for shadowing to perform a [consistent cutover](#consistent-cutover), along with a database replication technology that replicates writes to the target database.
```

1. Any writes that were made during the cutover will have been missed on CockroachDB. Use [MOLT Verify]({% link {{ page.version.version }}/molt-verify.md %}) to identify the inconsistencies. These will need to be manually reconciled.
{% endcomment %}

> ## Monitor cutover

Reviewer 3: Have you considered making this an H3 under the previous section? Maybe you decided against it to avoid having H4s, but it seems a little odd that you might monitor a cutover you didn't initiate.

Author: I did go back and forth on that -- not sure why I settled on an H2. I agree with you.

You can monitor your cutover attempts with the following commands.

> ### `connections list`

Reviewer 3: Did you mean to make these H4s or leave them at H3?

Author: 🤦 Good point. I'll fix in a follow-up PR.
